Src/HoneyBear.HalClient/HalClientAsyncExtensions.cs

```diff
@@ -0,0 +1,352 @@
+        {
+            var client = (HalClient) await hal;
+            return await client.ExecutePatchAsync(rel, value, parameters, curie);
+        }
+
+        /// <summary>
+        /// Makes a HTTP DELETE request to the given link relation on the most recently navigated resource.
+        /// </summary>
+        /// <param name="hal">The instance of <see cref="IHalClient"/> to extend.</param>
+        /// <param name="rel">The link relation to follow.</param>
+        /// <returns>The updated <see cref="IHalClient"/>.</returns>
+        /// <exception cref="AggregateException" />
+        public static Task<IHalClient> DeleteAsync(this Task<IHalClient> hal, string rel) =>
+            hal.DeleteAsync(rel, null, null);
+
+        /// <summary>
+        /// Makes a HTTP DELETE request to the given link relation on the most recently navigated resource.
+        /// </summary>
+        /// <param name="hal">The instance of <see cref="IHalClient"/> to extend.</param>
+        /// <param name="rel">The link relation to follow.</param>
+        /// <param name="curie">The curie of the link relation.</param>
+        /// <returns>The updated <see cref="IHalClient"/>.</returns>
+        /// <exception cref="AggregateException" />
+        public static Task<IHalClient> DeleteAsync(this Task<IHalClient> hal, string rel, string curie) =>
+            hal.DeleteAsync(rel, null, curie);
+
+        /// <summary>
+        /// Makes a HTTP DELETE request to the given templated link relation on the most recently navigated resource.
+        /// </summary>
+        /// <param name="hal">The instance of <see cref="IHalClient"/> to extend.</param>
+        /// <param name="rel">The templated link relation to follow.</param>
+        /// <param name="parameters">An anonymous object containing the template parameters to apply.</param>
+        /// <returns>The updated <see cref="IHalClient"/>.</returns>
+        /// <exception cref="AggregateException" />
+        public static Task<IHalClient> DeleteAsync(this Task<IHalClient> hal, string rel, object parameters) =>
+            hal.DeleteAsync(rel, parameters, null);
+
+        /// <summary>
+        /// Makes a HTTP DELETE request to the given templated link relation on the most recently navigated resource.
+        /// </summary>
+        /// <param name="hal">The instance of <see cref="IHalClient"/> to extend.</param>
+        /// <param name="rel">The templated link relation to follow.</param>
+        /// <param name="parameters">An anonymous object containing the template parameters to apply.</param>
+        /// <param name="curie">The curie of the link relation.</param>
+        /// <returns>The updated <see cref="IHalClient"/>.</returns>
+        /// <exception cref="AggregateException" />
+        public static async Task<IHalClient> DeleteAsync(this Task<IHalClient> hal, string rel, object parameters, string curie)
+        {
+            var client = (HalClient) await hal;
```

Comment:
I'm not really to fond of those castings, it feels "hacky". Is there a reason why you wouldn't want to include the async methods in the interface declaration.

Reply:
Yeah, the casts definitely smelled a bit.  As suggested, I've added the async methods to IHalClient.

Could you take a look, please?